iot_verifier/src/rewarder.rs

```diff
@@ -120,8 +189,72 @@ impl Rewarder {
             .await?
             .await??;
         self.reward_manifests_sink.commit().await?;
+        let sleep_duration = scheduler.sleep_until_next_epoch(Utc::now())?;
+        Ok(sleep_duration)
+    }
 
-        Ok(())
+    async fn threshold_check(
+        &self,
+        gateway_reward_shares: &GatewayShares,
+        scheduler: &Scheduler,
+    ) -> anyhow::Result<bool> {
+        let duration_since_last_rewards_end =
+            scheduler.time_since_last_reward_period_ended(Utc::now());
+        let halt_rewards_override = fetch_halt_rewards_override(&self.pool).await?;
+        // get the current averages from historic data in the db
+        let (scaled_beacon_avg, scaled_witness_avg, scaled_packet_avg) =
+            self.get_historic_averages().await?;
+
+        tracing::info!(%gateway_reward_shares.total_beacon_count,
+            %gateway_reward_shares.total_witness_count,
+            %gateway_reward_shares.total_packet_count,
```

Comment:
how does the "key = value" short-hand syntax treat these when they're struct fields being dot-accessed directly?